### PR TITLE
stalwart-mail: package und use upstream systemd unit

### PR DESCRIPTION
**Pull request by onny**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/mail/stalwart-mail.nix [hunk 1]
```diff
@@ -73,61 +73,56 @@ in {
...
-        UMask = "0077";
+    systemd = {
+      packages = [ cfg.package ];
+      services.stalwart-mail = {
+        wantedBy = [ "multi-user.target" ];
+        after = [ "local-fs.target" "network.target" ];
+        serviceConfig = {
+          ExecStart = [
+            ""
+            "${cfg.package}/bin/stalwart-mail --config=${configFile}"
```
Comment (SuperSandro2000) [inline nixos/modules/services/mail/stalwart-mail.nix:84]:
```suggestion
            "${lib.getExe cfg.package} --config=${configFile}"
```
we should set meta.mainProgram tbh

Comment (onny) [reply nixos/modules/services/mail/stalwart-mail.nix:84]: the package also contains a cli managing tool, so not sure if meta.mainProgram should be the cli-tool or server app

File: nixos/modules/services/mail/stalwart-mail.nix [hunk 2]
```diff
@@ -73,61 +73,56 @@ in {
...
+          ];
+
+          StandardOutput = "journal";
+          StandardError = "journal";
+
+          DynamicUser = true;
```
[the remark this answers is not in the document]
Comment (onny) [reply nixos/modules/services/mail/stalwart-mail.nix:90]: does it store that much on disk or does it use databases such as rocksdb for that?

[the remark this answers is not in the document]
Comment (oddlama) [reply nixos/modules/services/mail/stalwart-mail.nix:90]: if users choose sqlite it will probably store a lot of data in its directory.

**Review by pacien — APPROVED** [review]

**Review by pacien — APPROVED** [review]

**Comment by kurnevsky** [comment]
Was it intentional that the group/user in systemd are removed? Here it was added https://github.com/NixOS/nixpkgs/pull/315697 . Right now linux user is created but not used which seems wrong.

**Comment by kurnevsky** [comment]
Ah, ok, thanks. Turns out I was missing another thing - `AF_UNIX` in `RestrictAddressFamilies` . Postgresql can be used with unix sockets which I was trying to configure, but got confused with these users. Should it be included by default maybe?

**Comment by onny** [comment]
That would be nice, maybe toggle AF_UNIX depending on the configuration, if Postgresql is used? Would you like to open a pull request for it?

**Pull request merged**